src/components/CommentsList.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+import { fetchArticleComments } from "../../api";
+import { useEffect } from "react";
+import CommentCard from "./CommentCard";
+
+export default function CommentsList({ articleId }) {
+  const [comments, setComments] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
+    fetchArticleComments(articleId).then((comments) => {
+      setComments(comments);
+      setLoading(false);
+    });
+  }, [articleId]);
```

Comment:
The dependancy array does need to be here, however you don't need the articleId, this will only change if the page is navigated to a new article, where your article fetching will cause a re render on mount.